shell/store/__tests__/catalog.test.ts

```diff
@@ -0,0 +1,169 @@
+import { getters } from '../catalog';
+
+describe('getters', () => {
+  describe('chart', () => {
+    describe('should return nothing and avoid mismatches', () => {
+      it('if no parameter is passed', () => {
+        const result = (getters.chart as any)({}, {})({});
+
+        expect(result).toBeUndefined();
+      });
+
+      it.each([
+        {
+          repoType:  'repoType',
+          repoName:  'repoName',
+          chartName: undefined,
+        },
+        {
+          repoType:  undefined,
+          repoName:  'repoName',
+          chartName: 'chartName',
+        },
+        {
+          repoType:  'repoType',
+          repoName:  undefined,
+          chartName: 'chartName',
+        },
+      ])('if missing arguments for filtering %p', (options) => {
+        const result = (getters.chart as any)({}, {})({
+          key: undefined,
+          ...options
+        });
+
+        expect(result).toBeUndefined();
+      });
+    });
+
+    describe('should return latest version of matching chart for', () => {
+      it('given parameters', () => {
+        const options = {
+          chartName: 'chartName',
+          repoType:  'repoType',
+          repoName:  'repoName',
+        };
+        const common = { deprecated: false };
+        const chart1 = {
+          ...options,
+          ...common
+        };
+        const chart2 = {
+          ...options,
+          ...common,
+          repoName: 'somethingElse'
+        };
+        const state = {};
+        const stateGetters = { charts: [chart2, chart1] };
+
+        // NOTE: Getters arguments are actually optional
+        const result = (getters.chart as any)(state, stateGetters)(options);
+
+        expect(result).toStrictEqual(chart1);
+      });
+
+      it.each([
+        {
+          chartName: 'rancher-alerting-drivers',
+          repoType:  'cluster',
+          repoName:  'rancher-charts',
+        },
+        { chartName: 'rancher-alerting-drivers' },
+        { repoType: 'cluster' },
```

Comment:
given only repo type or repo name i don't think we should match any chart in that repo. feels like the minimum would be the chart name